Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -47,5 +56,34 @@ public async Task<ScheduleDTO> GetByDatePeriodForCurrentUserAsync(DateTime? date
             var schedules = await _scheduleRepository.GetByDatePeriodAsync(userId, date.Value, daysNum);
             return _mapper.Map<ScheduleDTO>(schedules);
         }
+
+        public async Task UpdateScheduleSlotsForCurrentUserAsync(IEnumerable<DateTime> dates)
+        {
+            var currentUserId = _userProvider.GetUserId();
+            var periodStartDate = dates.Min();
+            var periodFinishDate = periodStartDate.Date.AddDays(Constants.Week.CountDays - (int)periodStartDate.DayOfWeek + 1);
+            dates = dates.Where(dt => dt < periodFinishDate).ToList();
+
+            var scheduleSlotsByCurrentUser = await _scheduleRepository.GetScheduleSlotsOfDatePeriodAsync(currentUserId, periodStartDate, periodFinishDate);
+
+            var removedListDateTime = _scheduleSlotHelper.GetRemovedDateTimeInSheduleSlots(GetBulkDateTimeInScheduleSlots(scheduleSlotsByCurrentUser), dates);
+            if (removedListDateTime.Any())
+            {
+                var validationResult = await _validator.ValidateAsync(scheduleSlotsByCurrentUser.Where(s => removedListDateTime.Contains(s.AvailableTime)));
+                if (!validationResult.IsValid)
+                {
+                    throw new ValidationException(validationResult.Errors);
+                }
+            }
+
+            var newListDateTime = _scheduleSlotHelper.GetAddedDateTimeInSheduleSlots(GetBulkDateTimeInScheduleSlots(scheduleSlotsByCurrentUser), dates);
+
+            await _scheduleRepository.BulkUpdateScheduleSlotsAsync(currentUserId, newListDateTime, removedListDateTime);
+        }
+
+        private IEnumerable<DateTime> GetBulkDateTimeInScheduleSlots(IEnumerable<ScheduleSlot> scheduleSlots)
```

Comment:
We don't use bulk for get operations, just remove word bulk

Reply:
+